front/app/components/EventCard/DateBlocks.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { memo } from 'react';
+import moment from 'moment';
+
+// i18n
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from './messages';
+
+// styling
+import styled from 'styled-components';
+import { media, colors } from 'utils/styleUtils';
+
+const EventDateBlocks = styled.div`
+  flex: 0 0 75px;
+  width: 75px;
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+
+  ${media.smallerThanMinTablet`
+    flex: 0 0 60px;
+    width: 60x;
+  `}
+`;
+
+const Separator = styled.div`
+  height: 18px;
+
+  ${media.smallerThanMinTablet`
+    width: 15px;
+    height: auto;
+  `}
+`;
+
+const EventDateBlockWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+
+  &.second {
+    margin-top: 14px;
+  }
+`;
+
+const EventDateBlockLabel = styled.div`
+  color: ${colors.label};
+  font-size: 12px;
+  line-height: normal;
+  font-weight: 600;
+  text-transform: uppercase;
+  margin-bottom: 4px;
+`;
+
+const EventDateBlock = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+`;
+
+const EventDate = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: stretch;
+  padding: 6px;
+  border-radius: ${(props: any) => props.theme.borderRadius};
+  border-bottom-left-radius: 0;
+  border-bottom-right-radius: 0;
+  background: #f5f6f7;
+  border: solid 1px #ccc;
+  border-bottom: none;
+
+  ${media.smallerThanMinTablet`
+    padding: 4px;
+  `}
+`;
+
+const EventMonth = styled.div`
+  color: ${(props: any) => props.theme.colorText};
+  font-size: 14px;
+  line-height: normal;
+  font-weight: 500;
+  text-transform: uppercase;
+`;
+
+const EventDay = styled.div`
+  color: ${(props: any) => props.theme.colorText};
+  font-size: 17px;
+  line-height: normal;
+  font-weight: 400;
+
+  ${media.smallerThanMinTablet`
+    font-size: 16px;
+  `}
+`;
+
+const EventYear = styled.div`
+  color: #fff;
+  font-size: 16px;
+  line-height: normal;
+  font-weight: 400;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 6px;
+  border-radius: ${(props: any) => props.theme.borderRadius};
+  border-top-left-radius: 0;
+  border-top-right-radius: 0;
+  background: ${({ theme }) => theme.colorMain};
+
+  ${media.smallerThanMinTablet`
+    font-size: 14px;
+  `}
+`;
+
+interface Props {
+  startAtMoment: moment.Moment;
+  endAtMoment: moment.Moment;
+  isMultiDayEvent: boolean;
+}
+
+export default memo<Props>((props) => {
+  const { startAtMoment, endAtMoment, isMultiDayEvent } = props;
```

Comment:
100% nitpicking but I tend to prefer this deconstruction on the function arguments level.